library/bignum.c

```diff
@@ -1377,54 +1377,22 @@ mbedtls_mpi_uint mbedtls_mpi_core_mla( mbedtls_mpi_uint *d, size_t d_len,
     mbedtls_mpi_uint c = 0; /* carry */
     size_t excess_len = d_len - s_len;
 
-#if defined(MULADDC_HUIT)
-    for( ; s_len >= 8; s_len -= 8 )
-    {
-        MULADDC_INIT
-        MULADDC_HUIT
-        MULADDC_STOP
-    }
+    size_t steps_x8 = s_len / 8;
+    size_t steps_x1 = s_len & 7;
 
-    for( ; s_len > 0; s_len-- )
+    while( steps_x8-- )
     {
```

Comment:
Does this resolve https://github.com/Mbed-TLS/mbedtls/issues/1717, or are there further savings to be had?

Reply:
Well, we could always save more code size by doing no unrolling at all, but that would came at a performance code, at least for some platforms and some applications. I measured (code size, perf) 3 options on 3 platforms and my conclusion was that keeping 8-1 was a good compromise between code size, perf, and simplicity (there are platforms with optimized asm for 8 steps at once, so we need 8-1 for them, and it's simpler to have the same steps on all platforms).

We can always fine-tune things later, but my personal opinion is that most of the savings were had by removing the extra 16x loop, and we should leave it at that for now.